src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -1286,8 +1286,8 @@ class CommonVisitor : public AST::BaseVisitor<Struct> {
                                          args, StructType, loc);
             }
 
-            if (args.size() > 0 && args.size() !=  StructType->n_members) {
-                throw SemanticError("StructConstructor arguments do not match the number of struct members", loc);
+            if (args.size() > 0 && args.size() >  StructType->n_members) {
+                throw SemanticError("StructConstructor arguments are greater the number of struct members", loc);
```

Comment:
I would say:
"Struct constructor has more arguments than the number of struct members"